barretenberg/cpp/src/barretenberg/flavor/goblin_ultra.hpp

```diff
@@ -276,6 +278,119 @@ class GoblinUltraFlavor {
         };
         // The plookup wires that store plookup read data.
         auto get_table_column_wires() { return RefArray{ w_l, w_r, w_o }; };
+
+        /**
+         * @brief Construct sorted list accumulator polynomial 's'.
+         *
+         * @details Compute s = s_1 + η*s_2 + η²*s_3 + η³*s_4 (via Horner) where s_i are the
+         * sorted concatenated witness/table polynomials
+         *
+         * @param key proving key
+         * @param sorted_list_polynomials sorted concatenated witness/table polynomials
+         * @param eta random challenge
+         * @return Polynomial
+         */
+        void compute_sorted_list_accumulator(FF eta)
+        {
+            const size_t circuit_size = this->circuit_size;
+
+            auto sorted_list_accumulator = Polynomial{ circuit_size };
+
+            // Construct s via Horner, i.e. s = s_1 + η(s_2 + η(s_3 + η*s_4))
+            for (size_t i = 0; i < circuit_size; ++i) {
+                FF T0 = this->sorted_polynomials[3][i];
+                T0 *= eta;
+                T0 += this->sorted_polynomials[2][i];
+                T0 *= eta;
+                T0 += this->sorted_polynomials[1][i];
+                T0 *= eta;
+                T0 += this->sorted_polynomials[0][i];
+                sorted_list_accumulator[i] = T0;
+            }
+            this->sorted_accum = sorted_list_accumulator.share();
+        }
+
+        void compute_sorted_accumulator_polynomials(FF eta)
```

Comment:
mostly copied from instance, removing some unnecessary relation parameter and prover poly stuff